pkg/sql/inspect/index_consistency_check.go

```diff
@@ -247,20 +248,27 @@ func (c *indexConsistencyCheck) Start(
 	}
 
 	if indexConsistencyHashEnabled.Get(&c.flowCtx.Cfg.Settings.SV) && len(allColNames) > 0 {
-		match, hashErr := c.hashesMatch(ctx, allColNames, predicate, queryArgs)
-		if hashErr != nil {
-			// If hashing fails, we usually fall back to the full check. But if the
-			// error stems from query construction, that's an internal bug and shouldn't
-			// be ignored.
-			if isQueryConstructionError(hashErr) {
-				return errors.WithAssertionFailure(hashErr)
+		// The hash precheck uses crdb_internal.datums_to_bytes, which depends on
+		// keyside.Encode. Skip if any column type isn’t encodable (i.e. TSQUERY, etc.).
+		if !allColumnsDatumsToBytesCompatible(c.columns) {
+			log.Dev.Infof(ctx, "skipping hash precheck for index %s: column type not compatible with datums_to_bytes",
+				c.secIndex.GetName())
+		} else {
+			match, hashErr := c.hashesMatch(ctx, allColNames, predicate, queryArgs)
+			if hashErr != nil {
+				if isQueryConstructionError(hashErr) {
+					// If hashing fails and the error stems from query construction,
+					// that's an internal bug and shouldn't be ignored.
+					return errors.WithAssertionFailure(hashErr)
+				}
+				// For all other hash errors, log and fall back.
+				log.Dev.Infof(ctx, "hash precheck for index consistency did not match; falling back to full check: %v", hashErr)
```

Comment:
super nit: is it ever possible for `hashErr` to be nil, but `match` to be false? even if it's not possible now, it might be a good idea to move this log line to be after line 271 (or have a different log message there).